packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/SchemaUtil.java

```diff
@@ -223,7 +223,12 @@ public static IType removeModelFromParameter(RequestParameterLocation parameterR
             returnType = ClassType.BINARY_DATA;
         } else if (!(returnType instanceof PrimitiveType)) {
             if (type instanceof EnumType) {
-                returnType = ClassType.STRING;
+                boolean isPathOrURIParameter = RequestParameterLocation.PATH == parameterRequestLocation
+                    || RequestParameterLocation.URI == parameterRequestLocation;
+                boolean isEnumAsString = !(JavaSettings.getInstance().isDataPlaneClient() && isPathOrURIParameter);
+                if (isEnumAsString) { // do not change enum to string type for data-plane path/url parameter
+                    returnType = ClassType.STRING;
+                }
```

Comment:
Why `isPathOrURIParameter ` matters here? Can we just say "DPG (from swagger) is string; anything from TypeSpec is enum"?

Reply:
Trying to limit the scope to path/uri parameter only to reduce the impact, but using swagger/typespec to decide whether to make enum as string is simpler and more straightforward. I will try to use swagger/typespec to decide whether to make enum as string, and see the impact.

Reply:
I think just try in `ServiceClientMapper`, call this method if swagger, don't call it if typespec (there should be a subclass for typespec).

Reply:
If adding the condition to `ServiceClientMapper`, we will also need to add it to `ProxyParameterMapper`, otherwise generated proxy method will have issue, it will use string type instead of enum type. But if it is typespec, in `ProxyParameterMapper`, we also need to execute other logics except the enum part logic in `removeModelFromParameter`, e.g. below part will change body type to `BinaryData`
https://github.com/microsoft/typespec/blob/27d1069c252eda4988d221c0a04ca839e4fe1a20/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/SchemaUtil.java#L224-L225`


Adding the checking logics to `ProxyParameterMapper` might be a bit complex. That's why I added the logic in `SchemaUtil` currently.

Reply:
Ideally, proxy method parameter should still be String. This is consistent with DPG (that parameter be String and BinaryData).

Reply:
If there is real difficulty, I am fine doing keep the Enum on proxy method.

Just don't condition on path/uri (I think swagger can have path/uri on client parameter too).

Reply:
make sense, I'm trying to add the conversion logic to change enum type to string type and keep enum type in proxy method be string.

Reply:
After discussion, we aggreed to put the logics into the mappers and keep the type in protocol method be enum as well.